### PR TITLE
Add json serialization support to ICECredentialType

### PR DESCRIPTION
**Pull request by wdouglass**
#### Description
When serializing to/from json, the ICECredentialType should serialize as as a string instead of an enumeration.

### CODE REVIEW

**Comment by wdouglass** [comment]
marking draft until i figure out the lint/testing stuff

**Comment by wdouglass** [comment]
So, as usually happens, a small change turned into a bigger one. I was not expecting this to have so many commits!

Tests are passing, this is ready for review. Squash if you like!

File: configuration_test.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func TestConfiguration_getICEServers(t *testing.T) {
 
 func TestConfigurationJSON(t *testing.T) {
 	j := `{
-    "iceServers": [{"URLs": ["turn:turn.example.org"],
+    "iceServers": [{"urls": ["turn:turn.example.org"],
```
Comment (Sean-Der) [inline configuration_test.go:44]: @wdouglass Is this a breaking change? 

Is it possible we can accept `URLs` and `urls` so we don't break existing users?

Comment (wdouglass) [reply configuration_test.go:44]: I'm not sure how this test  ever worked -- if you look in iceserver.go, that field always serializes/deserializes to "urls" anyway -- this was a faulty test.

also, in the equivalent structure in "upstream" webrtc, the field is "urls"

**Review by Sean-Der — APPROVED** [review]

**Pull request merged**